### PR TITLE
Options initilization and check

### PR DESCRIPTION
**Pull request by mwestphal**
This initialize options in the constructor and also add a check flag when using set/get method

### CODE REVIEW

File: src/library/F3DOptions.cxx [hunk 1]
```diff
@@ -730,6 +730,8 @@ F3DOptions F3DOptionsParser::GetOptionsFromCommandLine(std::vector<std::string>&
 //----------------------------------------------------------------------------
 void F3DOptionsParser::ConvertToNewAPI(const F3DOptions& oldOptions, f3d::options* newOptions)
 {
+  f3d::options::initialize(*newOptions);
```
Comment (Meakk) [inline src/library/F3DOptions.cxx:733]: It should be in `options::F3DInternals` constructor I think. Any reason to call it explicitely?

File: src/library/f3d_options.cxx [hunk 2]
```diff
@@ -83,4 +84,72 @@ template double options::get<>(const std::string& name) const;
...
+  // General
+  options.set("quiet", false);
+  options.set("verbose", false);
+
+  // Loader/Loading
+  options.set("animation-index", 0);
+  options.set("geometry-only", false);
+  options.set("progress", false);
+  options.set("camera-index", 0);
+  options.set("color", std::vector<double>({ 1., 1., 1. }));
```
Comment (Meakk) [inline src/library/f3d_options.cxx:100]: Would be awesome to support `std::initializer_list` to be able to call `options.set("color", { 1., 1., 1. });`

File: src/library/F3DOptions.cxx (hunk 1 above)
Comment (mwestphal) [reply src/library/F3DOptions.cxx:733]: Maybe someone do not want to initialize ? currently F3D do not require using it.

File: src/library/f3d_options.cxx (hunk 2 above)
Comment (mwestphal) [reply src/library/f3d_options.cxx:100]: indeed, outside of the scope though

File: src/library/F3DOptions.cxx (hunk 1 above)
Comment (Meakk) [reply src/library/F3DOptions.cxx:733]: Then how to determine if a user specifies an option but it's not valid? The map should be initialized in the constructor imo, then any call to `set` should verify the key is already there.

Comment (mwestphal) [reply src/library/F3DOptions.cxx]: I think you proposed behavior is indeed what we want. If we need more flexibility we can improve that in the future

**Comment by mwestphal** [comment]
@Meakk please review

**Pull request merged**